### PR TITLE
rotation module improvement

### PR DESCRIPTION
**Pull request by larryngton2**
implements part of #3524

### CODE REVIEW

**Comment by larryngton2** [comment]
preview: https://streamable.com/ajucek

**Comment by 1zun4** [comment]
It seems weird though. Can't know if it's render smoothing or slow rotations.

**Comment by larryngton2** [comment]
> It seems weird though. Can't know if it's render smoothing or slow rotations.

there is a toggle for it sooo idk
its meant to be used for instant rotations anyways

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt [hunk 1]
```diff
@@ -41,11 +42,15 @@ import net.minecraft.util.Pair
 
 object ModuleRotations : Module("Rotations", Category.RENDER) {
 
-    val showRotationVector by boolean("ShowRotationVector", false)
+    private val showRotationVector by boolean("ShowRotationVector", false)
+    private val smoothRotations by boolean("SmoothRotations", false)
+    private val smoothingFactor by float("SmoothFactor", 0.15f, 0.1f..0.9f)
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt]: Then make sure this is in a toggleable configurable at least, so it's "Smooth Rotation" -> "Factor". Make sure it's "Rotation", not "Rotations", so it matches the "ShowRotationVector"

**Review by 1zun4 — CHANGES_REQUESTED** [review]
see comment

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt (hunk 1 above)
Comment (larryngton2) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt]: I'll work on it when I get on my pc

**Review by 1zun4 — APPROVED** [review]

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt [hunk 2]
```diff
@@ -41,11 +41,14 @@ import net.minecraft.util.Pair
 
 object ModuleRotations : Module("Rotations", Category.RENDER) {
 
-    val showRotationVector by boolean("ShowRotationVector", false)
+    private val showRotationVector by boolean("ShowRotationVector", false)
+    private val smoothRotations by boolean("SmoothRotation", false)
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt:45]: It is off by default 👍🏻

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt [hunk 3]
```diff
@@ -87,7 +90,28 @@ object ModuleRotations : Module("Rotations", Category.RENDER) {
...
         return current ?: server
     }
 
+    /**
+     * Rotation Smoothing
+     */
+    private fun smoothRotation(from: Rotation, to: Rotation): Rotation {
+        val diffYaw = to.yaw - from.yaw
+        val diffPitch = to.pitch - from.pitch
+        val smoothingFactor = 0.25f
```
Comment (superblaubeere27) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt:110]: smoothing factor is hardcoded as it should 👍🏻

**Review by superblaubeere27 — APPROVED** [review]
All possible complaints are already fixed

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt (hunk 3 above)
Comment (larryngton2) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt:110]: it should have been a slider inside a toggleableconfigurable but i lost 99.9% of my braincells trying to move all of this shit to a class

**Comment by larryngton2** [comment]
can we get even more approvals :pray:

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt (hunk 3 above)
Comment (superblaubeere27) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleRotations.kt:110]: bruh, just use a ToggleableConfigurable

**Pull request merged**